crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -292,6 +292,31 @@ use std::{cell::UnsafeCell, marker::PhantomData};
 ///
 /// # bevy_ecs::system::assert_is_system(my_system);
 /// ```
+///
+/// ## Non-Sync Queries
```

Comment:
These docs are great; add a copy of this to `Res` and `NonSend`?

We should also try to deduplicate these docs with the `include_str!` doc macro trick; these are subtle and critical enough that I really don't want that duplication.